wrappers/kn/build.gradle.kts

```diff
@@ -0,0 +1,224 @@
+        standardOutput = stdout
+    }
+    return stdout.toString().removePrefix("v").trim()
+}
+
+group = "io.github.zxing-cpp"
+version = getTagName()
+
+Properties().apply {
+    rootProject.file("local.properties").takeIf { it.exists() && it.isFile }?.let { load(it.reader()) }
+}.onEach { (key, value) ->
+    if (key is String) ext[key] = value
+}
+
+val hostOs = System.getProperty("os.name")
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+kotlin {
+    val androidTargets = {
+        listOf(
+            androidNativeArm32(),
+            androidNativeArm64(),
+            androidNativeX86(),
+            androidNativeX64(),
+        )
+    }
+    val appleTargets = {
+        listOf(
+            iosX64(),
+            iosArm64(),
+            iosSimulatorArm64(),
+            macosX64(),
+            macosArm64(),
+            watchosX64(),
+            watchosArm32(),
+            watchosArm64(),
+            watchosSimulatorArm64(),
+            tvosX64(),
+            tvosArm64(),
+            tvosSimulatorArm64(),
+        )
+    }
+    val linuxTargets = {
+        listOf(
+            linuxX64(),
+            linuxArm64(),
+        )
+    }
+    val windowsTargets = {
+        listOf(
+            mingwX64(),
+        )
+    }
+    val enabledTargetList = mutableListOf<KotlinNativeTarget>()
+    enabledTargetList.addAll(androidTargets())
+    enabledTargetList.addAll(linuxTargets())
+    enabledTargetList.addAll(windowsTargets())
+
+    applyDefaultHierarchyTemplate()
+
+    if (hostOs == "Mac OS X") enabledTargetList.addAll(appleTargets())
+
+    sourceSets {
+        val nonAndroidNativeTest by creating {
+            dependsOn(nativeTest.get())
+            dependencies {
+                implementation(libs.test.korlibs.korim)
+            }
+        }
+        if (hostOs == "Mac OS X") {
+            appleTest.get().dependsOn(nonAndroidNativeTest)
+        }
+        linuxTest.get().dependsOn(nonAndroidNativeTest)
+        mingwTest.get().dependsOn(nonAndroidNativeTest)
+    }
+}
+
+krossCompile {
+    libraries {
+        val zxingCpp by creating {
+            sourceDir = file("../../core").absolutePath
+            outputPath = ""
+            libraryArtifactNames = listOf("libZXing.a")
+
+            cinterop {
+                packageName = "zxingcpp.cinterop"
+                headers = listOf("$sourceDir/src/zxing-c.h")
+            }
+            cmake.apply {
+                val buildPath = project.layout.buildDirectory.dir("cmake").get().asFile.absolutePath +
+                        "/{projectName}/{targetName}"
+                configParams {
+                    buildDir = buildPath
+                }
+                configParams += (ModifiablePlatformEntriesImpl().apply {
+                    buildType = "Release"
+                    buildSharedLibs = false
+                } + CustomCMakeCacheEntries(
+                    mapOf(
+                        "BUILD_C_API" to "ON",
+                    )
```

Comment:
Fixed

Reply:
Give this a look: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/a1893f969c437f7f6e515ac73a41528172cc7705
I have no any idea on how to fix it on linux targets, do you have any good solution?

Reply:
You wrote "kotlin/native comes bundles with an gcc8 toolchain". I don't how this statement explains the observed behavior. To my understanding, kn does not have anything to do with how any platform/native library is build (using cmake in this case). It might be that kn has some c-compiler toolchain involved to be able to automatically generate the cinterop package but a) I thought this is based on llvm and b) the zxing-c.h does include the header that breaks the build. Maybe the issue is some outdated cross-compilation package on macos/brew? I did not dig into this, yet.

Reply:
As for my understanding, because K/N uses a set of toolchain(llvm, clang...) and a prebuilt sysroot downloaded from there server for every specified target to compile K/N codes (refers to: https://github.com/JetBrains/kotlin/blob/a38e64a7c85180fe456e73447d071efe603a1242/kotlin-native/konan/konan.properties#L413C1-L416C17). Once I dont use the same sysroot form them, there will be error linking static libraries. I dont know whether my cognition is correct, but using identical compiler and sysroot indeed fixes an error occurred in linking period.

Reply:
As soon as I saw that GCC8 only "partially supports" CXX20, I suspected it was its reason.

Reply:
Added: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/98dc751d0ed1bd669dca3db53544d4be082c179c